light_controller/light_controller.py

```diff
@@ -0,0 +1,195 @@
+        self._buttons[button] = irCode
+        if dump:
+            self.dumpButtons()
+
+    def dumpButtons(self):
+        """
+        Record All The Buttons!
+        Saves the buttons into the configured button file
+        """
+        with open(self._buttonsFile, 'wb') as fp:
+            json.dump(self._buttons, fp)
+
+    def recordButtons(self):
+        """
+        Record button presses from the remote
+        :return: SFA
+        """
+        print("Protocol Version:", self.toy.protocolVersion)
+        for button in sorted(self.buttonNames):
+            self.recordButton(button, dump=False)
+        self.dumpButtons()
+
+    def lightsOn(self):
+        """
+        Turn the lights on
+        :return:
+        """
+        self.toy.transmitCode(self._buttons['on'])
+
+    def lightsOff(self):
+        """
+        Turn the lights off
+        :return:
+        """
+        self.toy.transmitCode(self._buttons['off'])
+
+    def setColour(self, colour):
+        """
+        Change the light colours
+        :param colour: Colour name from :mod:`getColours`
+        :return: None
+        """
+        if colour in self._colours:
+            self.toy.transmitCode(self._buttons[colour])
+
+    def sendCommand(self, command):
+        """
+        Send a command
+        :param command: Command to send to the lights
+        :return: None
+        """
+        if command in self._commands:
+            self.toy.transmitCode(self._buttons[command])
+
+    def getVersion(self):
+        """
+        Get version information from the IR Toy
+
+        :returns: Hardware version, firmware version
+        """
+        return self.toy.getVersion()
+
+    @property
+    def colours(self):
+        return self._colours
+
+    @property
+    def commands(self):
+        return self._commands
+
+    @property
+    def protocolVersion(self):
+        return self.toy.protocolVersion
+
+
+if __name__ == '__main__':
+    changer = LightController('/dev/ttyACM0')
+    print(changer.protocolVersion)
+    print(changer.getVersion())
+    sys.stdout.flush()
+
+    print("Turning lights on")
+    changer.lightsOn()
+    changer.setColour('white')
+    time.sleep(1)
+    try:
+        done = False
+        while (not done):
+            done = True
+            # for colour in ('red', 'green', 'blue'):
+            #     print colour
+            #     changer.setColour(colour)
+            #     time.sleep(1)
+            #
+            # for colour in ('red2', 'green2', 'blue2'):
+            #     print colour
+            #     changer.setColour(colour)
+            #     time.sleep(1)
+
+            # for colour in ('orange', 'cyan', 'purple'):
+            #     print colour
+            #     changer.setColour(colour)
+            #     time.sleep(1)
+            #
+            # for colour in ('orange2', 'cyan2', 'violet'):
+            #     print colour
+            #     changer.setColour(colour)
+            #     time.sleep(1)
```

Comment:
Instead of leaving this as commented-out code, maybe we can move this somewhere where the user can ask to just run a specific "test suite", so to speak?